src/Raven.Server/Documents/DatabasesLandlord.cs

```diff
@@ -81,7 +81,8 @@ private void HandleClusterDatabaseChanged(string databaseName, long index, strin
                 if (record.Topology.RelevantFor(_serverStore.NodeTag) == false)
                     return;
 
-                if (record.Disabled)
+                if (record.Disabled ||
+                    record.DatabaseState.Equals(DatabaseStateStatus.RestoreInProgress))
```

Comment:
Let's use `==` instead of `Equals` here. It is safer, less possibility to get NRE if we will change `DatabaseState` to nullable in the future (probably not, but a good practice)